substrate-processor/src/chain.ts

```diff
@@ -147,28 +147,66 @@ export class Chain {
     async getStorage(blockHash: string, prefix: string, name: string, ...keys: any[]) {
         let item = this.getStorageItem(prefix, name)
         assert(item.keys.length === keys.length)
-        let req = sto.getNameHash(prefix) + sto.getNameHash(name).slice(2)
-        for (let i = 0; i < keys.length; i++) {
-            req += sto.getKeyHash(
-                item.hashers[i],
-                this.scaleCodec.encodeToBinary(item.keys[i], keys[i])
-            ).slice(2)
-        }
+        let req = sto.getNameHash(prefix) + sto.getNameHash(name).slice(2) + this.getKeysHash(item, ...keys)
         let res = await this.client.call('state_getStorageAt', [req, blockHash])
-        if (res == null) {
+        return this.decodeStorageValue(item, res)
+    }
+
+    async queryStorage(blockHash: string, prefix: string, name: string, keysArray: any[]) {
+        let item = this.getStorageItem(prefix, name)
+        let storageHash = sto.getNameHash(prefix) + sto.getNameHash(name).slice(2)
+        let query = new Array(keysArray.length)
+        for (let i = 0; i < keysArray.length; i++){
+            let keys = keysArray[i]
+            if (item.keys.length > 1) {
+                assert(item.keys.length === keys.length)
+                query[i] = storageHash + this.getKeysHash(item, ...keys)
+            } else {
+                query[i] = storageHash + this.getKeysHash(item, keys)
+            }
+        }
+        let res = await this.client.call('state_queryStorageAt', [query, blockHash])
+        let changes: [string, string][] = res[0].changes
+        let results = new Array(query.length)
+        if (changes.length === results.length) {
+            for (let i = 0; i < results.length; i++) {
+                results[i] = this.decodeStorageValue(item, changes[i][1])
+            }
+        } else {
+            for (let i = 0; i < results.length; i++) {
+                let value = changes.find(change => change[0] === query[i])?.[1]
```

Comment:
`state_queryStorageAt` returns values whithout dups and nulls, so it is necessary to make it consistent with keys